inc/os_linux.h

```diff
@@ -7,15 +7,36 @@
 //
 // Linux OS api used in PAL implementation
 //
+
+// Include default apis ...
+#if !defined(_DEFAULT_SOURCE)
 #define _DEFAULT_SOURCE 1
 #define _BSD_SOURCE 1
 #define _SVID_SOURCE 1
+#endif
+
+// ... plus xopen and posix apis 
+#if !defined(_XOPEN_SOURCE) || (_XOPEN_SOURCE < 600)
+#undef _XOPEN_SOURCE
 #define _XOPEN_SOURCE 600
+#endif
+
+// ... and any linux specific ones
+#if !defined(_GNU_SOURCE)
 #define _GNU_SOURCE 1
+#endif
+
+// Ensure safe string manipulation ...
+#if !defined(_FORTIFY_SOURCE)
 #define _FORTIFY_SOURCE 1
+#endif
+
+// ... and C'99 compliant source
+#if !defined(_ISOC99_SOURCE)
 #define _ISOC99_SOURCE 1
-#include <features.h>
+#endif
 
+#include <features.h>
```

Comment:
As http://man7.org/linux/man-pages/man7/feature_test_macros.7.html#NOTES says,
features.h should not be included directly.
It will be included indirectly by some of the headers included by the following os_posix.h.